digest/src/lib.rs

```diff
@@ -0,0 +1,37 @@
+use multihash::Multihash;
+
+pub fn digest(bytes: &[u8]) -> Multihash {
+    // Time the tests and take fastest hash?
+    multihash::Sha3_256::digest(bytes)
+}
+
+pub trait DigestRoot {
+    fn digest_root(self) -> Multihash;
+}
+
+pub trait DigestField {
+    fn digest_field(self, field_name: String) -> Multihash;
+}
+
+impl DigestField for String {
```

Comment:
With the above discussion on noun/verb usage of 'digest', perhaps we could re-name this `FieldDigest` - then its a noun.